modules/lazyload/module.php

```diff
@@ -38,38 +38,125 @@ private function enqueue_styles() {
 		);
 	}
 
+	private function get_repeaters_selectors( $controls, $settings ) {
+		$selectors = [];
+		$repeater = array_filter( $controls, function( $control ) {
+			if ( 'repeater' === $control['type'] && Utils::get_array_value_by_keys( $control, [ 'fields', 'background_image', 'fields_options', 'image', 'background_lazyload', 'active' ] ) ) {
+				return true;
+			}
+		} );
+		if ( empty( $repeater ) ) {
+			return false;
+		}
+		$repeater = array_shift( $repeater );
+
+		if ( $repeater ) {
+			$repeater_settings = $settings[ $repeater['name'] ][0];
+			$repeater_selector = $this->get_repeater_selector( $repeater_settings, $repeater );
+			if ( $repeater_selector ) {
+				$selectors[] = $repeater_selector;
+			}
+		}
+
+		return $selectors;
+	}
+
+	private function get_repeater_selector( $repeater_settings, $repeater ) {
+		$lazyload_options = Utils::get_array_value_by_keys( $repeater, [ 'fields', 'background_image', 'fields_options', 'image', 'background_lazyload' ] );
+		if ( ! $lazyload_options['active'] ) {
+			return false;
+		}
+
+		$repeater_selector = Utils::get_array_value_by_keys( $lazyload_options, [ 'selector' ] );
+		if ( ! $repeater_selector ) {
+			return false;
+		}
+
+		$repeater_keys = Utils::get_array_value_by_keys( $repeater_settings, [ 'background_lazyload', 'keys' ] );
+		$repeater_background_image_url = Utils::get_array_value_by_keys( $repeater_settings, $repeater_keys );
+
+		if ( $repeater_background_image_url ) {
+			return $repeater_selector;
+		}
+
+		return false;
+	}
+
 	private function update_element_attributes( Element_Base $element ) {
```

Comment:
Maybe consider doing it like this:
```php
$settings = $element->get_settings_for_display();
$controls = $element->get_controls();

$selectors = [];

foreach ( $controls as $control_name => $control_data ) {
	if (
		$is_repeater // Here should be the check if the control is repeater.
	) {
		// You dont need to run over all the controls again, because you have the control data already.
		$selectors = array_merge(
			$this->get_repeater_selector( $control_data ),
			$selectors // Merge all teh selectors to one array.
		);

		// Because it is a repeater, you already handle it and you should not continue the code.
		continue;
	}

	// Here should be the code the normal control and get its selectors.
	// In this method if there is no selector specified, it will return the wrapper selector.
	$selectors = array_merge(
		$this->get_control_selector( $control_data ),
		$selectors // Merge all teh selectors to one array.
	);
}

// Here it should make the array_unique and trim and merge theme into one string (which is the one selector)
$final_selector = $this->merge_selectors( $selectors );
if ( ! empty( $final_selector ) ) {
	$element->add_render_attribute(
		'_wrapper',
		[ 'data-e-bg-lazyload' => $final_selector ]
	);
}
```

This is only a guideline,  I think there will be more changes you will have to do.

In my opinion it will make the code more readable, let me know what do you think?